src/pylint_datetime/__init__.py

```diff
@@ -1,4 +1,185 @@
+
+# function calls to fromisoformat, strptime and fromisoformat can't be checked - they parse strings
+# if the specified timezone is None, still allowed although they produce naive objects
+
+
+class DatetimeChecker(BaseChecker):  # type: ignore
+    """class for custom pylint checker."""
+
+    name = "pylint-datetime"
+    priority = -1
+    msgs = {
+        "W9999": (
+            "timedelta() called without keyword arguments",
+            "timedelta-no-keyword-args",
+            "timedelta() called without keyword arguments.",
+        ),
+        "W9998": (
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+            "timezone-no-argument",
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+        ),
+        "W9997": (
+            'Function call to "%s" can only produce naive objects',
+            "naive-datetime-call",
+            'Function call to "%s" can only produce naive objects, aware objects necessary',
+        ),
+        "W9996": (
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+            "naive-datetime-no-timezone",
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+        ),
+    }
+
+    def __init__(self, linter: pylint.lint.PyLinter = None) -> None:
+        """Initialize checker.
+
+        Args:
+            linter: instance of pylint linter.
+        """
+        super().__init__(linter)
+
+    def timedelta_with_keywords(
+        self, node: astroid.node_classes.NodeNG, name: str
+    ) -> None:
+        """Check calls to timedelta() and datetime.timedelta() for keyword arguments.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name == "timedelta":
+            for _ in node.args:
+                self.add_message("timedelta-no-keyword-args", node=node)
+            if not node.args and not node.keywords:
+                self.add_message("timedelta-no-keyword-args", node=node)
+
+    # check for positional arguments cheesy, uses fact, that they have one Attribute Argument.
+
+    def timezone_argument(self, node: astroid.node_classes.NodeNG, name: str) -> None:
+        """Check calls to datetime, now, fromtimestamp, astimezone and time for timezone arg.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name in ("datetime", "now", "fromtimestamp", "astimezone"):
+            if not any(
+                isinstance(arg, astroid.Keyword) and arg.arg in ("tz", "tzinfo")
+                for arg in node.keywords
+            ):
+                if not any(isinstance(arg, astroid.Attribute) for arg in node.args):
+                    self.add_message("timezone-no-argument", node=node, args=(name,))
```

Comment:
Does this still allows `datetime.now(timezone.utc)`, right? I'm not sure if `astroid.Attribute` covers for this or not.

Reply:
yes, it does allow it. that is what i meant with the positional check being a bit cheesy

Reply:
Ah, OK, I never seen *cheesy* used like this, sorry!